src/Compilers/CSharp/Test/Semantic/Semantics/QueryTests.cs

```diff
@@ -2379,5 +2379,91 @@ public static void Main(string[] args)
                 Diagnostic(ErrorCode.ERR_ExpressionVariableInQueryClause, "z").WithLocation(22, 55)
                 );
         }
+
+        [WorkItem(15910, "https://github.com/dotnet/roslyn/issues/15910")]
+        [Fact]
+        public void ExpressionVariablesInQueryClause_03()
+        {
+            var csSource = @"
+using System.Linq;
+
+class Program
+{
+    public static void Main(string[] args)
+    {
+        var a = new[] { (1, 2), (3, 4) };
+        var za = from x in M(a, (int qa, int wa) = a[0]) select x; // scoping ok
```

Comment:
I'm surprised. Weren't we discussing in another thread how deconstruction declarations are presently blocked (except as a statement)? Shouldn't `qa` and `qi` (below) report errors?